artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/server/impl/RemotingServiceImpl.java

```diff
@@ -345,10 +338,14 @@ public synchronized void startAcceptors() throws Exception {
       if (isStarted()) {
          for (Acceptor a : acceptors.values()) {
             try {
-               if (a instanceof NettyAcceptor acceptor && !acceptor.isAutoStart()) {
+               if (!a.isAutoStart()) {
                   continue;
                }
                a.start();
+
+               if (a instanceof NettyAcceptor nettyAcceptor && server.getMetricsManager() != null) {
```

Comment:
you could do a similar cleanup to the previous one here.


Create a method on the acceptor such as registerMetrics,

and then change this to


```java
if (server.getMetricsManager() != null) {
     a.registerMetrics(server);
}
```